mdtraj/formats/pdb/pdbfile.py

```diff
@@ -161,6 +164,8 @@ def load_pdb(filename, stride=None, atom_indices=None, frame=None,
 
         topology = f.topology
         if atom_indices is not None:
+            #to avoid modifying the topology if given as input
+            topology = deepcopy(topology)
             topology = topology.subset(atom_indices)
```

Comment:
This does very undefined stuff if you give it a pre-subsetted topology:
```python
>>> import mdtraj as md
>>> top = md.Topology()
>>> c = top.add_chain()
>>> r = top.add_residue('test', c, 0, 0)
>>> top.add_atom('a0', 'H', r, 0)
test0-a0
>>> top.add_atom('a1', 'H', r, 1)
test0-a1
>>> top
<mdtraj.Topology with 1 chains, 1 residues, 2 atoms, 0 bonds at 0x7fa5a7447090>
>>> tops = top.subset([1])
>>> tops
<mdtraj.Topology with 1 chains, 1 residues, 1 atoms, 0 bonds at 0x7fa5445f3490>
>>> tops2 = tops.subset([1]) # Redo the same subset
>>> tops2
<mdtraj.Topology with 0 chains, 0 residues, 0 atoms, 0 bonds at 0x7fa5445f35d0>
```
Maybe error out if `topology.n_atoms` is not equal to `len(atom_indices)` after this `subset`?